Source/DafnyLanguageServer/MarkupTestFile.cs

```diff
@@ -61,7 +61,7 @@ public static class MarkupTestFile {
 
       var r = new Regex(@"(?<Position>><)|" +
                         @"(?<SpanStart>\[>)|(?<SpanEnd><\])" +
-                        @"|(?<NameSpanStart>\{>(?<Name>[-_.A-Za-z0-9\+]+)\:)|(?<NameSpanEnd><\})" +
+                        @"|(?<NameSpanStart>\{>(?<Name>[-_.'=(){}""A-Za-z0-9\*\+]+)\:)|(?<NameSpanEnd><\})" +
```

Comment:
What is that? You don't use this for a test, do you?

Reply:
New Language server tests were failing because this regex did not include special symbols that were present. This regex captures the text of the quick fix label; sometimes that label includes the text of what is to be removed or inserted -- that can often have special symbols in it. (e.g. replace '=' with ':=')

Reply:
Could you please point me to "New Language server tests"? I don't see any new language server tests in this PR.

Reply:
And normally, you'd use regular spans, not namespans.

Reply:
There are no new Language server tests in this PR. I tried to add some but continually ran into the stability problem that Rustan reported. I would still like to add them, but that stability problem has to be fixed first.

Reply:
"And normally, you'd use regular spans, not namespans." -- are you suggesting a change? I'm happy to accommodate a more appropriate use of the LS test infrastructure.

Reply:
I thought there was another comment from you on this thread. But I can't find it back. To clarify:

The Language server tests are not very stable. Probably 50% of the time they fail for me on win or macos and have to be rerun. This is independent of the work on parser errors.

When I tried adding more LS tests, even simple ones, the instability rapidly became unusable. So given my limited time, I focussed on the work you have reviewed, with manual tests of the actions. I'd like to add LS tests of them but it will have to wait until the instability in general is fixed. I'd say that the data point of adding more tests increasing instability supports Remy's hypothesis that the problem has to do with concurrency.

As for the regex you asked about. The label for a Quick Fix is given in the test markup as the text between `(<` and `->` -- so any text that does not include `->` should be OK as a label. And indeed, it allows for clear labels like `replace '=' by ':='`

Reply:
> The Language server tests are not very stable. Probably 50% of the time they fail for me on win or macos and have to be rerun. This is independent of the work on parser errors.

I don't experience that. I haven't followed the context very well, so I don't know if this applies, but whatever the circumstances are we shouldn't merge code that isn't automatically tested, something we did previously in #3299, because that creates a large risk of breaking user facing features later.

If for some reason we can't add the tests we would like to add, that needs to be resolved before we continue.

Reply:
I do have better success when I'm running late at night -- though that seems so odd I have trouble believing it.

Note also that I can add a few tests without too much trouble, but adding lots of them gives a high failure rate.

For now I can split the PR into one with the error explanations and one with the code fixes to be added later -- which likely then is a long time later

Reply:
I conferred with Robin: he proposes that we merge what we have, with the quick fixes, and I put my not yet merged code with quick fix tests in a separate PR, pending work on resolving flaky LS tests. And if this PR with additional tests helps to do that, all the better.

Reply:
Based on what you say, I assume that reverting this line would make the tests of this PR to fail, although I still don't understand why this line matter.

Reply:
No - it is needed for the LS tests that are not included here because of the instability previously discussed. So it is not needed for this PR, but will be needed in the future. Again - the reason it is needed is to accommodate more than simple alpha-numeric text in labels of quick fixes, like saying `replace '=' with ':='`